monitoring/monitors/viya/podMonitor-sas-go-pods.yaml

```diff
@@ -22,11 +22,32 @@ spec:
     #   regex: golang
     #   action: keep
     - sourceLabels: [__meta_kubernetes_pod_label_app_kubernetes_io_name]
-      regex: "sas-arke|sas-launcher"
+      regex: "sas-arke|sas-launcher|sas-redis-operator"
       action: drop
     - sourceLabels: [__meta_kubernetes_pod_annotation_sas_com_kustomize_base]
       regex: "golang"
       action: keep
+    - action: replace
+      regex: true;(.+)
+      replacement: $1
+      sourceLabels:
+      - __meta_kubernetes_pod_annotationpresent_prometheus_io_port
+      - __meta_kubernetes_pod_annotation_prometheus_io_port
+      targetLabel: __port
+    - action: replace
+      regex: (.+);(.+)
+      replacement: $1:$2
+      sourceLabels:
+      - __meta_kubernetes_pod_ip
+      - __port
+      targetLabel: __address__
+    - action: replace
+      regex: true;(.+)
+      replacement: $1
+      sourceLabels:
+      - __meta_kubernetes_pod_annotationpresent_prometheus_io_path
```

Comment:
I went ahead and removed this section because it does not affect the functionality of the scraping - I'm thinking it might've been a customization that they did for their own deployments.

Reply:
I suspect this is related to the OpenSearch service running w/in SAS Viya.  The user is running an older version of SAS Viya.  In that version of SAS Viya, the OpenSearch instance w/in SAS Viya is monitored using an instance of the Elasticsearch Exporter...which makes metrics available on the endpoint /metrics rather than the /internal/metrics endpoint used by the SAS Golang-based pods.  

This block of code would tell the monitor to refer to the annotations to figure out the endpoint.  So, for the Elasticsearch Exporter pod, the annotations would tell it to use "/metrics" instead of "/internal/metrics".  Which is correct; and will likely eliminate this pod from showing up in Prometheus as "DOWN". So, problem solved!  EXCEPT the Elasticsearch Exporter isn't really one of the intended targets of this monitor (i.e. it is NOT an SAS-authored, Golang microservice).  Therefore,  I think we should handle this pod the same way we handle the other things that get "caught" by this monitor but we don't want to monitor; e.g. redis, sas-launcher, etc.  So, let's add "sas-opendistro-exporter" to that exclusion list.